app/src/main/java/org/mifos/mobile/ui/transaction/TransactionActivity.kt

```diff
@@ -0,0 +1,87 @@
+package org.mifos.mobile.ui.transaction
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_transaction.*
+import org.json.JSONObject
+import org.mifos.mobile.R
+import org.mifos.mobile.models.entity.*
+import org.mifos.mobile.ui.activities.base.BaseActivity
+import org.mifos.mobile.utils.Toaster
+import javax.inject.Inject
+
+class TransactionActivity : BaseActivity(), TransactionContract.View, View.OnClickListener {
+
+    @Inject
+    internal lateinit var transactionPresenter: TransactionPresenter
+    var jsonob: JSONObject? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_transaction)
+        showBackButton()
+        setToolbarTitle("Send Money")
+        activityComponent?.inject(this)
+        transactionPresenter.attachView(this)
+        send.setOnClickListener(this)
+        if(intent.getStringExtra("JSON")!=null) {
```

Comment:
can you explain your logic here?

Reply:
This is the `JSON` bundle. Here we are basically checking that only.